subprojects/plugins/src/integTest/groovy/org/gradle/api/tasks/JavaExecIntegrationTest.groovy

```diff
@@ -80,6 +83,30 @@ class JavaExecIntegrationTest extends AbstractIntegrationSpec {
         executedAndNotSkipped ":run"
     }
 
+    def 'arguments can be passed via command line and take precedence'() {
+        when:
+        run("run", "--args", "2 '3' \"4\"")
+
+        then:
+        executedAndNotSkipped ":run"
+        assertOutputFileIs('''\
+        2
+        3
+        4
+        '''.stripIndent())
+
+        when:
+        run("run", "--args", "2 '3' \"4\"")
+
+        then:
+        executedAndNotSkipped ":run"
+        assertOutputFileIs('''\
+        2
+        3
+        4
+        '''.stripIndent())
```

Comment:
Why do we run this twice?

Reply:
To test it's not marked as `UP-TO-DATE`.

Reply:
Why isn't it up-to-date here but in the test below?

Reply:
Nice catch, because some output is registered via API: `run.outputs.file "out.txt"`.

Reply:
I see. Thanks for the pointer! Can we make this more clear in the test name?

Reply:
Sure. Thanks for reminding!